### PR TITLE
Created Range algorithm

### PR DESCRIPTION
**Pull request by silardi**
I created an algorithm that detects range expansion and contraction and makes decisions to trade based on the concept that range expansion indicates trend continuation and range contraction indicates trend reversal. It performed pretty well against Chester's algorithm with the stocks and time periods that I included in main.py. I'm very curious to see how it will perform when backtesting over the dates for our six different regimes once we determine what they will be. 

Feel free to let me know if you guys have any questions.

### CODE REVIEW

File: range/main.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        # self.ticker = "^GSPC"
+        self.ticker = "TQQQ"
+        # self.ticker = "TNA"
+        # self.ticker = "^IXIC"
+        # self.ticker = "UPRO"
+        # self.ticker = "UDOW"
+        # self.ticker = "SOXL"
+
+    def Initialize(self):
+        get_yahoo_data(self.ticker, '2015-01-01', '2022-04-23')
```
Comment (scheffeltravis) [inline range/main.py:24]: Is there a reason we pull data starting from 2015, when we set the start date below as the beginning of 2017?

**Review by scheffeltravis — APPROVED** [review]
Looks awesome man! Great job figuring out how to get the range contraction/expansion detection working.

File: range/main.py (hunk 1 above)
Comment (silardi) [reply range/main.py:24]: I think I was testing 2015 start dates at some point but deleted them from the commented out dates. I kept the date range larger when backtesting so I didn't need to worry about changing it when I wanted to test an older date.

**Review by jhdaguitera — APPROVED** [review]

**Comment by jhdaguitera** [comment]
Looks good!

**Pull request merged**